programs/perpetuals/src/state/pool.rs

```diff
@@ -243,7 +243,10 @@ impl Pool {
         };
         let close_amount = max_collateral_price
             .get_token_amount(available_amount_usd, collateral_custody.decimals)?;
-        let max_amount = math::checked_add(position.locked_amount, position.collateral_amount)?;
+        let max_amount = math::checked_add(
+            math::checked_sub(position.locked_amount, fee_amount)?,
```

Comment:
I think potentially fee_amount can be greater than locked_amount. To be 100% safe lets use saturating_sub here and in assets.owned calc above

Reply:
yes, agree with you!